src/Common/FitOnFhir.Common/Interfaces/IAuthStateService.cs

```diff
@@ -22,5 +22,22 @@ public interface IAuthStateService
         /// Will throw an <see cref="ArgumentException"/> if <see cref="AuthenticationConfiguration.IsAnonymousLoginEnabled"/> is false and
         /// query params <see cref="Constants.ExternalIdQueryParameter"/> and <see cref="Constants.ExternalSystemQueryParameter"/> are present in the <see cref="HttpRequest"/>.</returns>
         AuthState CreateAuthState(HttpRequest httpRequest);
```

Comment:
I am seeing that the `CreateAuthState()` method seems more like an internal member. Since the AuthState object is only used externally for the Callback request, It might make sense to encapsulate the creation of a new object in the AuthStateService.

Reply:
Never mind. I see now that you are using this for the `Revoke()` flow too.